src/main/java/com/ipcounter/App.java

```diff
@@ -0,0 +1,50 @@
+package com.ipcounter;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import com.ipcounter.classes.FileProcessor;
+
+public class App {
+  public static void main(String[] args) {
+    Scanner sc = new Scanner(System.in);
+    boolean finishedCounting = false;
+
+    // lets the user change the path after a mistake
+    while (!finishedCounting) {
+      System.out.println("Please provide a path to the file with the list of IPs:");
+      String path = sc.nextLine();
+      try {
+        // creating a separate thread
+        FileProcessor processor = new FileProcessor(path);
+        System.out.println("Starting the counting process, please wait.");
+        // running the process in a separate thread so we can show the progress
+        processor.start();
+        float lineNumber = 0;
+        final int arbitraryBeautifulNumber = 1000;
+        // while the counting continues we display the progress for every 1000 steps
+        while (processor.isAlive()) {
+          float updatedLineNumber = processor.getCurrentLine();
+          if (updatedLineNumber > lineNumber &&
+              updatedLineNumber % arbitraryBeautifulNumber == 0) {
+            lineNumber = updatedLineNumber;
+            System.out.println(lineNumber);
```

Comment:
if file contains 4*10^9 entries, something about 4 millions lines will be showed and it is huge task for console.